core/src/main/scala/codesearch/core/db/repository/PackageIndexDbRepository.scala

```diff
@@ -0,0 +1,57 @@
+package codesearch.core.db.repository
+
+import cats.Monad
+import cats.implicits._
+import doobie._
+import doobie.implicits._
+import fs2.Stream
+
+final case class PackageIndexTableRow(
+    name: String,
+    version: String,
+    repository: String
+)
+
+final case class PackageIndex(
+    name: String,
+    version: String
+)
+
+trait PackageIndexRepository[F[_]] {
+  def batchUpsert(packages: List[PackageIndexTableRow]): F[Int]
+  def batchUpsert(stream: Stream[F, PackageIndexTableRow]): F[Int]
+  def findLatestByRepository(repository: String): Stream[F, PackageIndexTableRow]
+}
+
+object PackageIndexDbRepository {
+  def apply[F[_]: Monad](xa: Transactor[F]): PackageIndexRepository[F] = new PackageIndexRepository[F] {
+
+    def batchUpsert(packages: List[PackageIndexTableRow]): F[Int] = {
+      Update[PackageIndexTableRow](
+        """
+          |INSERT INTO repository_index(name, version, repository)
```

Comment:
What are the pipes doing here?

Reply:
@neongreen it's Scala multi-row strings definition syntax.